convert2rhel/pkghandler.py

```diff
@@ -458,6 +459,11 @@ def replace_non_rhel_installed_kernel(version):
 
     pkg = "kernel-%s" % version
 
+    utils.remove_pkgs(['kmod-kvdo', 'libreport'], critical=False)
+    for pkgname in set(system_info.pkg_blacklist):
+
+        utils.install_pkgs([pkgname.replace("centos", "redhat"), pkgname.replace("centos", "rhel")], replace=True)
+
     ret_code = utils.download_pkg(
         pkg=pkg, dest=utils.TMP_DIR, disablerepo=tool_opts.disablerepo,
         enablerepo=tool_opts.enablerepo)
```

Comment:
After the kernel rpm is downloaded, call a new function that would check if there are any missing deps of the downloaded kernel rpms and install them. Something like:
```
install_missing_kernel_deps(kernel_rpm_path)
```

Then the new function could look like:
```
def install_missing_kernel_deps(kernel_rpm_path):
    """Since we use rpm for force-replacing the kernel, we need to make sure that all the required dependencies
    of the downloaded kernel rpm are installed. Otherwise the force replacement would fail.
    """
    yb = yum.YumBase()
    yb.reinstallLocal(pkg=kernel_rpm_path)
    yb.resolveDeps()
    pkgs_in_transaction = yb.tsInfo.getMembers()
    to_install = []
    for pkg in pkgs_in_transaction:
        if pkg.reason == "dep":
            to_install.append(pkg.name)
    if to_install:
        loggerinst = logging.getLogger(__name__)
        loggerinst.info("Installing missing kernel dependencies before the kernel force-replacement.")
        _, ret_code = call_yum_cmd("install", "%s" % (" ".join(to_install)))
        if ret_code != 0:
            loggerinst.critical("Unable to install kernel dependencies: %s", % (", ".join(to_install)))
```